consensus/core/src/leader_schedule.rs

```diff
@@ -39,6 +47,118 @@ impl LeaderSchedule {
+        let leader_swap_table = dag_state.read().last_reputation_scores_from_store().map_or(
+            LeaderSwapTable::default(),
+            |reputation_scores| {
+                LeaderSwapTable::new(
+                    context.clone(),
+                    reputation_scores,
+                    context
+                        .protocol_config
+                        .consensus_bad_nodes_stake_threshold(),
+                )
+            },
+        );
+        // create the schedule
+        Self::new(context, leader_swap_table)
+    }
+
+    pub(crate) fn commits_until_leader_schedule_update(
+        &self,
+        dag_state: Arc<RwLock<DagState>>,
+    ) -> usize {
+        let unscored_committed_subdags_count = dag_state.read().unscored_committed_subdags_count();
+        assert!(
+            unscored_committed_subdags_count <= self.num_commits_per_schedule,
+            "Unscored committed subdags count exceeds the number of commits per schedule"
+        );
+        self.num_commits_per_schedule
+            .saturating_sub(unscored_committed_subdags_count) as usize
+    }
+
+    pub(crate) fn update_leader_schedule(
+        &self,
+        dag_state: Arc<RwLock<DagState>>,
+        committer: &UniversalCommitter,
+    ) {
+        let _s = self
+            .context
+            .metrics
+            .node_metrics
+            .scope_processing_time
+            .with_label_values(&["LeaderSchedule::update_leader_schedule"])
+            .start_timer();
+
+        let mut dag_state = dag_state.write();
+        let unscored_subdags = dag_state.take_unscored_committed_subdags();
+
+        // TODO: remove this once scoring strategy is finalized
+        let scoring_strategy =
+            if let Ok(scoring_strategy) = std::env::var("CONSENSUS_SCORING_STRATEGY") {
+                tracing::info!(
+                    "Using scoring strategy {scoring_strategy} for ReputationScoreCalculator"
+                );
+
+                let scoring_strategy: Box<dyn ScoringStrategy> = match scoring_strategy.as_str() {
+                    "vote" => Box::new(VoteScoringStrategy {}),
+                    "certified_vote_v1" => Box::new(CertifiedVoteScoringStrategyV1 {}),
+                    "certified_vote_v2" => Box::new(CertifiedVoteScoringStrategyV2 {}),
+                    "certificate" => Box::new(CertificateScoringStrategy {}),
+                    _ => Box::new(VoteScoringStrategy {}),
+                };
+                scoring_strategy
+            } else {
+                tracing::info!(
+                    "Using scoring strategy VoteScoringStrategy for ReputationScoreCalculator"
+                );
+                Box::new(VoteScoringStrategy {})
+            };
+
+        let score_calculation_timer = self
+            .context
+            .metrics
+            .node_metrics
+            .scope_processing_time
+            .with_label_values(&["ReputationScoreCalculator::calculate"])
```

Comment:
Do you see different between the two scopes? It seems the computation difference between the two scopes is small.

Reply:
Very slight differences, but will need to do some more thorough testing on private testnet and then can consolidate these scopes if I am not seeing much value.

[example](https://metrics.sui.io/goto/PMzCuAYSR?orgId=1)